data-raw/mockdata.R

```diff
@@ -0,0 +1,459 @@
+# ------------------------------------------------------------------------------
+
+panel_AB_T1_T2_long <- dplyr::bind_rows(
+  panel_A_T1_long,
+  panel_B_T1_long,
+  panel_A_T2_long,
+  panel_B_T2_long
+)
+
+
+# sanity check if the sample sizes are allright
+panel_AB_T1_T2_long |>
+  dplyr::count(group, time, subject_id) |>
+  dplyr::count(group, time)
+
+# ------------------------------------------------------------------------------
+# 7) Add counts and fold_change + unique sample_id's
+# ------------------------------------------------------------------------------
+
+# additionally to the binary 0's and 1's we usually also have the raw number of
+# reads, denoted here as counts_control and counts_hits; it is important to know
+# that the reads you see in the original phipseq data are reads only for the
+# peptides, which were marked as "hits" in the normalization pipelines. You
+# don't really see all the other missing peptides from the library
+#
+# Simple sampling from Poisson + pseudocounts for fold_change. Should be fine
+# for now; may expand/refine in the future
+withr::with_preserve_seed({
+  panel_AB_T1_T2_long <- panel_AB_T1_T2_long |>
+    dplyr::mutate(
+      sample_id = droplevels(factor(sample_id)),
+      lambda_ctrl = runif(dplyr::n(), min = 0, max = 30),
+      lambda_hits = runif(dplyr::n(), min = 10, max = 5000),
+      counts_control = rpois(dplyr::n(), lambda = lambda_ctrl),
+      counts_hits = rpois(dplyr::n(), lambda = lambda_hits),
+      fold_change = (counts_hits + 0.5) / (counts_control + 0.5)
+    ) |>
+    dplyr::select(-lambda_ctrl, -lambda_hits)
+})
+# ------------------------------------------------------------------------------
+# 8) Final touches + save the data
+# ------------------------------------------------------------------------------
+# convert the subject_id and peptide_id to characters
+panel_AB_T1_T2_long$subject_id <- as.character(panel_AB_T1_T2_long$subject_id)
+panel_AB_T1_T2_long$peptide_id <- as.character(panel_AB_T1_T2_long$peptide_id)
+
+# remove potential duplicates
+panel_AB_T1_T2_long <- panel_AB_T1_T2_long |>
+  dplyr::arrange(subject_id, time, peptide_id) |>
+  dplyr::distinct(subject_id, time, peptide_id, .keep_all = TRUE)
+
+# Saving:
+# it is a little bit tricky, because the phip_convert() functions do not support
+# converting a raw data.frame. Maybe i should add this function but usually the
+# data are so big, you can not really store them in a data.frame, so it made no
+# sense for me at the beginning
+#
+# Therefore i will save the data in inst/extdata and in the R/resolve-paths.R a
+# small helper was defined, to help the user get the absolute path to the
+# dataset without messing with the package installation paths
+
+out_path <- file.path("inst", "extdata", "phip_mixture.parquet")
+arrow::write_parquet(panel_AB_T1_T2_long, out_path)
+
+# reproducibility check
+all.equal(current_seed, .Random.seed)
+
+# The usage would be then:
```

Comment:
I think this part can be moved to the README (but maybe at a later time if you want to re-write parts of the readme anyway...)